### ISSUE TITLE
BUG: Private Practitioners have the option to file Entry of Appearance documents across a consolidated group if they are associated with at least one other case in the group

### ISSUE THREAD

**Issue by ttlenard** [ustaxcourt/ef-cms#8551]
**Describe the Bug**
If a private practitioner requests access to a case in a consolidated group, and they are NOT yet associated with any other cases in the group, they do NOT have the option to file the entry of appearance document in all cases in the group (Good)

If a private practitioner requests access to a case in a consolidated group and the ARE associated with at least one other case in the group already, they DO have the option to file the entry of appearance document in all case in the group (BAD)

Per story ustaxcourt/ef-cms#8535 (see comment from Tom 4/27) Private practitioners should not have the ability to file Entry of Appearance documents across a consolidated group.

**Business Impact/Reason for Severity**
Medium

**In which environment did you see this bug?**
Staging, Test

**Who were you logged in as?**
Private Practitioner

**What were you doing when you discovered this bug? (Using the application, demoing, smoke tests, testing other functionality, etc.)**
Testing Other Functionality

**To Reproduce**
1. Log in as a private practitioner that isn't associated with any cases in a consolidated group
2. Search for a case docket number of a consolidated case
3. Click request access to case button
4. Select any of the entry of appearance options
5. There is NO option to file across a group
6. File the entry of appearance document so that you get associated with the case
7. Next, from your dashboard, find another case docket number from that consolidated group
8. Type that docket number in the search bar, and go to that case
9. Click request access to case
10. Select any of the Entry of appearance type documents (with the exception of the auto-generated doc)
11. Notice that the private practitioner has the option to file the EA across the group.  This shouldn't be there!

**Expected Behavior**
Private Practitioners should not be able to file EA type documents in all cases in the consolidated group

**Actual Behavior**
A clear and concise description of what actually happened.
Right now, if a private practitioner is already associated with at least one other case in a consolidated group, then they will have the option to file the EA type document in all cases in the group

**Screenshots**

This user is associated with 2 cases in a consolidated group:
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN2tNQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d77a398d9b92e4e3679c3c6ea2f38c644eb55476/image.png)

When requesting access to this other case in the group, they have the option to file in all cases in the group.  This shouldn't be there!
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN2dNQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--fe4557422f2f01aece1a14cc324e3c171aa6f95f/image.png)

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Cause of Bug, If Known**

**Process for Logging a Bug:**
* Complete the above information
* Add a severity tag (Critical, High Severity, Medium Severity or Low Severity). See below for priority definition. 

**Severity Definition:**
* Critical Defect
Blocks entire system's or module’s functionality
No workarounds available
Testing cannot proceed further without bug being fixed.

* High-severity Defect
Affects key functionality of an application
There's a workaround, but not obvious or easy
App behaves in a way that is strongly different from the one stated in the requirements

* Medium-severity Defect
A minor function does not behave in a way stated in the requirements.
Workaround is available and easy

* Low-severity Defect
Mostly related to an application’s UI
Doesn't need a workaround, because it doesn't impact functionality

## Definition of Ready for Bugs(Created 10-4-21)
Definition used: A failure or flaw in the system which produces an incorrect or undesired result that deviates from the expected result or behavior. (Note: Expected results are use cases that have been documented in past user stories as acceptance criteria and test cases, and do not include strange behavior unrelated to use cases.)

The following criteria must be met in order for the development team to begin work on the bug.

The bug must:
 - Be focused on solving a user problem
 - Contain data for all fields in the bug template, so the team can pick it up and begin working immediately

Process: If the unexpected results are new use cases that have been identified, but not yet built, new acceptance criteria and test cases should be captured in a new user story and prioritized by the product owner. 

If the Court is not able to reproduce the bug, add the “Unable to reproduce” tag. This will provide visibility into the type of support that may be needed by the Court. In the event that the Court cannot reproduce the bug, the Court will work with Flexion to communicate what type of troubleshooting help may be needed.

## Definition of Done (Updated 4-14-21)
**Product Owner**
 - [ ]  Bug fix has been validated in the Court's test environment

**Engineering**
 - [x] Automated test scripts have been written
 - [x] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [x] Verify that language for docket record for internal users and external users is identical
 - [x] New screens have been added to pa11y scripts
 - [x] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [x] READMEs, other appropriate docs and swagger/APIs fully updated
 - [x] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [x] Interactors should validate entities before calling persistence methods
 - [x] Code refactored for clarity and to remove any known technical debt
 - [ ] Deployed to the Court's `test` environment if prod-like data is required. Otherwise, deployed to any `experimental` environment for review.

**Comment by rachelschneiderman** [comment]
- Can the feature flag for external multi-docketing be removed?
- Can we unify case association multi-docketing permissions? (both private and IRS practitioners will NOT be able to multi-docket case association requests)
- Do we need to change the list of multi-docketable event codes to be specific to e-filing events NOT paper filed events?

**Comment by ttlenard** [comment]
Note: With this Fix, we've removed the ability for IRS Practitioners to file an EA across a consolidated group.  I had pulled the data in Kibana and found that there were 0 instances of this actually occurring since we had originally deployed that functionality.  This fix made the functionality consistent b/t private practitioners and IRS, where nobody can file an EA across a consolidated group.

Testing passes!  Thanks @Absolutestunna @rachelschneiderman @zRogersFlexion
